packages/pg/lib/connection-parameters.js

```diff
@@ -125,9 +125,25 @@ class ConnectionParameters {
   }
 
   getLibpqConnectionString(cb) {
-    var params = []
-    add(params, this, 'user')
-    add(params, this, 'password')
+    if (typeof this.password === 'function') {
+      const pw = this.password();
+      if (typeof pw === 'string') {
+        return this._getLibpqConnectionString(cb, pw);
+      }
+      pw.then(pwString => this._getLibpqConnectionString(cb, pwString));
```

Comment:
This probably needs to handle rejections.

Reply:
Yep, fixed!